src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryDate.cs

```diff
@@ -374,27 +509,29 @@ public static FormulaValue Second(IRContext irContext, FormulaValue[] args)
 
         // https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-date-time
         // Date(Year,Month,Day)
-        public static FormulaValue Date(IRContext irContext, NumberValue[] args)
+        public static FormulaValue Date(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, NumberValue[] args)
         {
             // $$$ fix impl
             var year = (int)args[0].Value;
             var month = (int)args[1].Value;
             var day = (int)args[2].Value;
 
-            return DateImpl(irContext, year, month, day);
+            return DateImpl(runner, context, irContext, year, month, day);
         }
 
-        private static FormulaValue DateImpl(IRContext irContext, int year, int month, int day)
+        private static FormulaValue DateImpl(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, int year, int month, int day)
         {
             // The final date is built up this way to allow for inputs which overflow,
             // such as: Date(2000, 25, 69) -> 3/10/2002
             try
             {
-                var result = new DateTime(year, 1, 1)
+                var datetime = new DateTime(year, 1, 1)
                     .AddMonths(month - 1)
                     .AddDays(day - 1);
 
-                return new DateValue(irContext, result);
+                datetime = MakeValidDateTime(runner, datetime, runner.GetService<TimeZoneInfo>() ?? LocalTimeZone);
+
+                return datetime.Hour == 0 ? new DateValue(irContext, datetime) : new DateTimeValue(irContext, datetime);
```

Comment:
>return datetime.Hour == 0 ? new DateValue(irContext, datetime) : new DateTimeValue(irContext, datetime); [](http://example.com/codeflow?start=16&length=104)

This doesn't look right. DateTime(2022,9,1,0,0,0) should return a DateTimeValue - the definition of the DateTime function requires it to return a DateTime. Date(2021,3,28) in Beirut should return a Date value (so that if we pass it to the Text function, it should not print out the time component)